### PR TITLE
Support operators in CodeGenerator

### PR DESCRIPTION
**Pull request by kevmal**
Addresses issue #206

Shadows `(|SpecificCall|_|)` in `CodeGenerator` to match on target methods. Otherwise 
https://github.com/fsprojects/FSharp.TypeProviders.SDK/blob/592d23a96db79b4248ed23a062db3a17433282ce/src/ProvidedTypes.fs#L13893-L13906
never matches.

Still some missing operators and more testing is needed (particularly for unsigned ints).

Certain operators don't have a `NoDynamicInvocation` such as (+) and (*) but are still included here. Ops so far:
```
~-
~+
-
+
*
/
% 
|||
^^^
~~~
&&&
int32/int
float/double
float32/single
int16
int64
byte
sbyte
uint32
uint16
uint64
```

@dsyme Any general feedback on this before continuing?

### CODE REVIEW

**Comment by dsyme** [comment]
The general direction is looking good

See also https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1071-witness-passing-quotations.md which would mean we don't have to do this thing, as FSharp.Core would have callable entrypoints for all operators.

File: src/ProvidedTypes.fs [hunk 1]
```diff
@@ -13889,6 +13961,518 @@ namespace ProviderImplementation.ProvidedTypes
...
+                | String ->
+                    ilg.Emit(I_call(Normalcall, (convTypeToTgt typeof<System.String>).GetMethod("CompareOrdinal", [|t1; t1|]) |> transMeth, None))
+                    emitExpr ExpectedStackState.Value <@@ 0 @@>
+                    ilg.Emit(I_clt)
+                | StaticMethod "op_LessThan" [|t1; t1|] m -> 
+                    ilg.Emit(I_call(Normalcall, transMeth m, None))
+                | _ -> failwithf "Operator (<) not supported for type %s" t1.Name
+
+           
+            | SpecificCall <@ (>) @>(None, [t1], [a1; a2]) ->
```
Comment (dsyme) [inline src/ProvidedTypes.fs]: In general, for performance reasons, each of these uses of `SpecificCall` should be lifted to a helper pattern, defined at the module level, e.g.

```
let (|GreaterThan|_|) = (|SpecificCall|_|) <@ (>) @>
```
This prevents the repeated creation of a quotation for `<@ (>) @>` and, IIRC, the implementation of SpecificCall does some pre-processing on the partial application.

Comment (kevmal) [reply src/ProvidedTypes.fs]: updated

**Comment by dsyme** [comment]
@kevmal This looks ready to go?

**Comment by kevmal** [comment]
This should be good for now. 

Some of what's not included here:
- hash/ compare/ pown though they should still work through FSharp.Core calls.
- Operators.Checked -- could still get exceptions from some of these
- isNull -- I may need to handle this at some point (works fine)
- reraise -- This will not work and will need to be supported at some point

Included the System.Math ops which could also moved out of CodeGenerator since most don't emit specific IL.

**Comment by dsyme** [comment]
Thank you!!

**Pull request merged**